library/Icinga/Chart/Primitive/Circle.php

```diff
@@ -61,8 +62,22 @@ public function toSvg(RenderContext $ctx)
         $circle->setAttribute('cx', Format::formatSVGNumber($coords[0]));
         $circle->setAttribute('cy', Format::formatSVGNumber($coords[1]));
         $circle->setAttribute('r', $this->radius);
-        $circle->setAttribute('style', $this->getStyle());
+
+        $this->id = $this->id ?? uniqid('circle-');
```

Comment:
Please set the id like this in all styleable classes you've changed.
```suggestion
        $this->setId($this->id ?? uniqid('circle-'));
```